cmd/soroban-rpc/internal/events/events.go

```diff
@@ -207,6 +207,7 @@ func (m *MemoryStore) IngestEvents(ledgerCloseMeta xdr.LedgerCloseMeta) error {
 	}
 	m.lock.Lock()
```

Comment:
instead we can go with `defer m.lock.Unlock() `

Reply:
I think his idea is to unlock asap but I can't imagine deferring would make a significant performance difference for two `Observe` calls (cc @sreuland who had same comment)

Reply:
Yeah no performance diff but I prefer it that way so that further error branches dont have to worry about unlocking it. I think we have the similar pattern in [here](https://github.com/stellar/soroban-rpc/blob/main/cmd/soroban-rpc/internal/transactions/transactions.go#L124) in transaction.go [Anything works as long as we unlock it :) ]

Reply:
agreed!

Reply:
not with the same granularity